native/acter/src/api/news.rs

```diff
@@ -165,24 +166,49 @@ impl NewsSlide {
 
     pub fn has_formatted_text(&self) -> bool {
         matches!(
-            self.inner.content(),
-            NewsContent::Text(TextMessageEventContent {
+            self.inner.content().text(),
+            Some(TextMessageEventContent {
                 formatted: Some(_),
                 ..
-            })
+            }),
         )
     }
 
     pub fn text(&self) -> String {
         match self.inner.content() {
-            NewsContent::Image(ImageMessageEventContent { body, .. }) => body.clone(),
-            NewsContent::Audio(AudioMessageEventContent { body, .. }) => body.clone(),
-            NewsContent::Video(VideoMessageEventContent { body, .. }) => body.clone(),
-            NewsContent::File(FileMessageEventContent { body, .. }) => body.clone(),
-            NewsContent::Location(LocationMessageEventContent { body, .. }) => body.clone(),
+            NewsContent::Image(ImageMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Image(ImageMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
+            NewsContent::File(FileMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::File(FileMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
+            NewsContent::Location(LocationMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Location(LocationMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
+            NewsContent::Video(VideoMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Video(VideoMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
+            NewsContent::Audio(AudioMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Audio(AudioMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
```

Comment:
aren't all these `body` of the same type? Could be one match in that case... but also we might want to consider moving that into `core` as it is a bit ... verbose... What do you think?

Reply:
Yes all cases have body as ```&String```. Could have possibility of moving in core, let me do it.